### ISSUE TITLE
Account sequence increments even when the blob is not submitted successfully

### ISSUE THREAD

**Issue by a26nine** [rollkit/rollkit#1787]
### Version of Git SHA

rollkit version:  `v0.13.5`
rollkit git sha:  `6a33192`

### System OS

Ubuntu 22.04

### Steps to reproduce it

It happens randomly on the rollup node, there are no specific steps to reproduce it.
- The rollup node tries to submit blobs through the light node
- There `context deadline` is `exceeded`
- It throws `incorrect account sequence` error on the next try because the account sequence has been incremented

### Expected result

If the blob submission in the earlier attempt was unsuccessful, it should be retried with the same sequence/nonce.

### Actual result

```
ERR DA layer submission failed error="failed to submit blocks: RPC client error: sendRequest failed: Post \"http://localhost:56658/\": context deadline exceeded" attempt=0 module=BlockManager
ERR DA layer submission failed error="failed to submit blocks: RPC client error: sendRequest failed: Post \"http://localhost:56658/\": context deadline exceeded" attempt=1 module=BlockManager
ERR DA layer submission failed error="failed to submit blocks: tx failed with code 32: account sequence mismatch, expected 600, got 601: incorrect account sequence\n: incorrect account sequence" attempt=2 module=BlockManager
```

I have seen the account sequence discrepancy of upto 30 sequences.

### Notes

In such case, one needs to manually post the blobs to sync the actual and expected sequence number.

**Comment by tuxcanfly** [comment]
Tried to repro but seems to be fixed in celestia-node >= v0.13.7

See: https://github.com/celestiaorg/celestia-node/issues/3164#issuecomment-2163232709

**Comment by a26nine** [comment]
> Tried to repro with `v0.15.1-rc0` but seems to be fixed in `celestia-node v0.13.7`
> 
> See: [celestiaorg/celestia-node#3164 (comment)](https://github.com/celestiaorg/celestia-node/issues/3164#issuecomment-2163232709)

Okay, I am on `v0.14.x` now and haven't encountered this yet. Let's close this for now.
